src/sentry/static/sentry/app/components/strictClick.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+
+/**
+ * Usage:
+ *   <StrictClick onClick={this.onClickHandler}>
+ *     <button>Some button</button>
+ *   </StrictClick>
+ */
+const StrictClick = React.createClass({
+  propTypes: {
+    onClick: React.PropTypes.func.isRequired
+  },
+
+  mixins: [
+    PureRenderMixin
+  ],
+
+  statics: {
+    MAX_DELTA_X: 10,
+    MAX_DELTA_Y: 10
+  },
+
+  getInitialState() {
+    return {
+      startCoords: null
+    };
+  },
+
+  handleMouseDown: function(evt) {
+    this.setState({
+      startCoords: {
+        x: evt.screenX,
+        y: evt.screenY
+      }
+    });
+  },
+
+  handleMouseClick: function(evt) {
+    // Click happens if mouse down/up in same element - click will
+    // not fire if either initial mouse down OR final ouse up occurs in
+    // different element
+    let {startCoords} = this.state;
+    let deltaX = evt.screenX - startCoords.x;
+    let deltaY = evt.screenY - startCoords.y;
+
+    // If mouse hasn't moved more than 10 pixels in either Y
+    // or X direction, fire onClick
+    if (deltaX < StrictClick.MAX_DELTA_X && deltaY < StrictClick.MAX_DELTA_Y) {
+      this.props.onClick(evt);
+    }
+    this.setState({
+      startCoords: null
+    });
+  },
+
+  render() {
+    return React.cloneElement(this.props.children, {
```

Comment:
If this is cloning the children, does this end up with two layers in DOM? An `<ol>` inside an `<ol>`? Or am I just misunderstanding?